### PR TITLE
Add setup instructions for burn-tch

### LINKED ISSUE #587: Comprehensive Documentation Needed for libtorch Setup and Troubleshooting
### Issue Description

In the process of building and running our Burn projects, it has been observed that some users are encountering difficulties with setting the environment variables and the CUDA version for libtorch. Additionally, some are experiencing issues when executing a built binary with Rust where the libtorch library path is lost, likely due to it being outside of the Cargo context.

Although libtorch is downloaded during the build, these complications highlight the need for a more detailed and thorough documentation page that addresses libtorch setup and troubleshooting.

### Suggested Solution

We need to create a comprehensive document that covers all aspects related to libtorch setup, including but not limited to:

- Setting environment variables
- Selecting and configuring the correct CUDA version
- Navigating issues with the libtorch library path, particularly in relation to the Cargo context
- Document should be written not only from the burn's developer perspective but also from the end user perspective who use Burn framework as a library. 

This document should also include a troubleshooting guide that helps users diagnose and resolve common problems they may encounter during setup or operation.

### Additional Information

By improving our documentation, we can better support our users and potentially reduce the number of support queries related to libtorch setup. Detailed, clear, and accurate documentation is a critical component of user experience and can play a significant role in the success of our projects.

If there are any suggestions or feedback about what to include in this documentation, please comment on this issue.

### PR DESCRIPTION
**Pull request by laggui**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Closes #587

### Changes

This draft PR adds detailed instructions to setup the `tch` backend with LibTorch and a simple example to test that the installation is successful.

Because many different installation configurations are possible, I'd love to get some feedback on the recommended method.

At this time I switched to a manual installation with proper documentation since the automatic download doesn't work for all use cases (e.g., MPS).

Maybe we still want to offer automatic by default and add separate that as a feature to allow manual installation support?

### CODE REVIEW

**Comment by laggui** [comment]
@nathanielsimard @antimora would love to get your opinion

**Comment by laggui** [comment]
Just chatted with Nathaniel, the suggestion is:

1. Keep automatic download by default
2. Set the correct tch environment variables based on feature flags (cpu (default), cu118 and cu121 for current libtorch version)
  - Still undecided if this is the best practice or other approaches would be better
4. Provide manual instructions for other use cases (as currently added in the documentation)

**Review by antimora — CHANGES_REQUESTED** [review]
Thank you for working on this. It's a great start and I am sure, we will extend it as we encounter issues. 

Can you also add Troubleshooting section. Often people report ATen errors which is related to incorrectly installing CUDA version. You might be able to fine this from https://github.com/tracel-ai/burn/issues/587 comments. 

As I recommend we keep auto downloader, which also works for Mac. The manual configuration is a recommended path.

File: burn-tch/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ burn-tensor = { path = "../burn-tensor", version = "0.12.0" }
 half = { workspace = true, features = ["std"] }
 libc = "0.2.150"
 rand = { workspace = true, features = ["std"] }
-tch = { version = "0.14.0", features = ["download-libtorch"] }
+tch = { version = "0.14.0" }
```
[the remark this answers is not in the document]
Comment (laggui) [reply burn-tch/Cargo.toml:23]: Yes agreed. We agreed to keep automatic installation as a default, was going to make the changes today.

File: burn-tch/README.md [hunk 2]
```diff
@@ -6,49 +6,208 @@
...
+export DYLD_LIBRARY_PATH=/path/to/pytorch/lib:$DYLD_LIBRARY_PATH
+```
+
+## Example Usage
+
+For a simple example, check out [simple-add.rs](examples/simple-add.rs). It automatically selects
+the device for your installation and performs a simple elementwise addition.
+
+For a more complete example using the `tch` backend, take a loot at the
+[Burn mnist example](https://github.com/tracel-ai/burn/tree/main/examples/mnist).
```
[the remark this answers is not in the document]
Comment (laggui) [reply burn-tch/README.md:213]: Ahhh yes that makes sense! Will do.

**Comment by laggui** [comment]
> Thank you for working on this. It's a great start and I am sure, we will extend it as we encounter issues.
> 
> Can you also add Troubleshooting section. Often people report ATen errors which is related to incorrectly installing CUDA version. You might be able to fine this from #587 comments.
> 
> As I recommend we keep auto downloader, which also works for Mac. The manual configuration is a recommended path.

We're back to automatic download now! :) And a flag `burn-tch/custom-libtorch` can be specified for a manual/custom libtorch installation.

Sadly, there is no way to "automatically" set the `TORCH_CUDA_VERSION` environment variable for the `tch` dependency.. `[env]` is only for the cargo config and the dependencies are built before so there's no way to "hack" this in a `build.rs`. Would have been nice to avoid this for our users, but instead I've added instructions for the automatic download w/ CUDA (which basically only requires setting the `TORCH_CUDA_VERSION` beforehand).

Regarding the troubleshooting section, I'm not sure what should be included. The CUDA driver version compatibility is already mentioned is the instructions and no CUDA toolkit is actually required (contrary to what some issues seem to refer to). For a manual installation direct download links are provided along with instructions to set the proper environment variables that would otherwise cause issues. Do you mean adding troubleshooting for even more custom installations (i.e., using a local LibTorch version that is not the currently listed version supported by `tch`)?

File: burn-core/Cargo.toml [hunk 3]
```diff
@@ -60,6 +60,7 @@ blas-netlib = ["burn-ndarray?/blas-netlib"]
 
 ndarray = ["burn-ndarray"]
 tch = ["burn-tch"]
+tch-custom = ["burn-tch?/custom-libtorch"]
```
[the remark this answers is not in the document]
Comment (laggui) [reply burn-core/Cargo.toml:63]: Oh right! The worst part is I stumbled upon that last Friday but it completely slept my mind today 🤦 

Thanks 🙏  will fix

**Review by louisfd — APPROVED** [review]
Looks good to me on a superficial level, haven't delved into all the details. There's only one typo

File: burn-tch/README.md [hunk 4]
```diff
@@ -6,49 +6,255 @@
...
+cargo run --bin cuda --release
+cargo run --bin mps --release
+```
+
+_Note: no MPS distribution is available for automatic download at this time, please check out the
+[manual instructions](#metal-mps)._
+
+### Manual Download
+
+To install `tch-rs` with a different LibTorch distribution, you will have to set the to manually
```
[the remark this answers is not in the document]
Comment (laggui) [reply burn-tch/README.md]: Whoops, good catch!

Leftover from a previous commit where the wording was different. Removed "to set the".

**Pull request merged**